src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,194 @@
+package duke.parser;
+
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Handles user input and commands.
+ */
+public class Parser {
+
+    private boolean isExit = false;
+    private Ui ui = new Ui();
+
+    /**
+     * Handles user command and give an appropriate reply.
+     * @param userInput commands specified by the user.
+     */
+    public void parse(String userInput) {
+        String[] input = userInput.split(" ", 2);
+
+        Type type = getEnumType(input[0]);
+        TaskList taskList = new TaskList();
+
+        try {
```

Comment:
I really like how readable this block is, as well as how you printed different types of error messages for different types of errors :smiley:

Reply:
Thank you! 😄